pkg/driverbuilder/builder/archlinux.go

```diff
@@ -36,59 +36,93 @@ func (c *archlinux) TemplateScript() string {
-				archKernelVersion,
-				archKernelVersion,
-				kr.Fullversion,
-				customVersion,
-				kr.Architecture.ToNonDeb())
-		case strings.Contains(kr.Extraversion, "-zen"):
-			archKernelVersion = "-zen"
-			customVersion = strings.ReplaceAll(kr.Extraversion, archKernelVersion, "")
-			baseurl = fmt.Sprintf("https://archive.archlinux.org/packages/l/linux%s-headers/linux%s-headers-%s.%s-%s.pkg.tar",
-				archKernelVersion,
-				archKernelVersion,
-				kr.Fullversion,
-				customVersion,
-				kr.Architecture.ToNonDeb())
-		default:
-			baseurl = fmt.Sprintf("https://archive.archlinux.org/packages/l/linux-headers/linux-headers-%s.%s-%s.pkg.tar",
-				kr.Fullversion,
-				kr.Extraversion,
-				kr.Architecture.ToNonDeb())
+	// check the architecture, which limits the mirror options
+	if kr.Architecture.ToNonDeb() == "x86_64" {
+		if strings.Contains(kr.FullExtraversion, "arch") { // arch stable kernel
+			baseURL := "https://archive.archlinux.org/packages/l/linux-headers"
+			for _, compressionAlgo := range possibleCompressionSuffixes {
+				urls = append(
+					urls,
+					fmt.Sprintf(
+						"%s/linux-headers-%s%s-%s.pkg.tar.%s",
+						baseURL,
+						kr.Fullversion,
+						kr.FullExtraversion,
+						kr.Architecture.ToNonDeb(),
+						compressionAlgo,
+					),
+				)
+			}
+		} else if strings.Contains(kr.FullExtraversion, "hardened") || strings.Contains(kr.FullExtraversion, ".a-1") { // arch hardened kernel ("a-1" is old naming standard)
+			baseURL := "https://archive.archlinux.org/packages/l/linux-hardened-headers"
+			for _, compressionAlgo := range possibleCompressionSuffixes {
+				urls = append(
+					urls,
+					fmt.Sprintf(
+						"%s/linux-hardened-headers-%s%s-%s.pkg.tar.%s",
+						baseURL,
+						kr.Fullversion,
+						kr.FullExtraversion,
+						kr.Architecture.ToNonDeb(),
+						compressionAlgo,
+					),
+				)
+			}
+		} else if strings.Contains(kr.FullExtraversion, "zen") { // arch zen kernel
+			baseURL := "https://archive.archlinux.org/packages/l/linux-zen-headers"
+			for _, compressionAlgo := range possibleCompressionSuffixes {
+				urls = append(
+					urls,
+					fmt.Sprintf(
+						"%s/linux-zen-headers-%s%s-%s.pkg.tar.%s",
+						baseURL,
+						kr.Fullversion,
+						kr.FullExtraversion,
+						kr.Architecture.ToNonDeb(),
+						compressionAlgo,
+					),
+				)
+			}
+		} else { // arch LTS kernel
+			baseURL := "https://archive.archlinux.org/packages/l/linux-lts-headers"
+			for _, compressionAlgo := range possibleCompressionSuffixes {
+				urls = append(
+					urls,
+					fmt.Sprintf(
+						"%s/linux-lts-headers-%s%s-%s.pkg.tar.%s",
+						baseURL,
+						kr.Fullversion,
+						kr.FullExtraversion,
+						kr.Architecture.ToNonDeb(),
+						compressionAlgo,
+					),
+				)
+			}
+		}
+	} else if kr.Architecture.ToNonDeb() == "aarch64" {
+		baseURL := "http://tardis.tiny-vps.com/aarm/packages/l/linux-aarch64-headers/"
```

Comment:
This is an odd mirror

Reply:
I agree! See here: https://github.com/falcosecurity/kernel-crawler/pull/61#issuecomment-1297220299

Neither Federico or I could find a better spot for `aarch64` hosted Arch kernel headers... Federico had been using this one previously, so I kept it on the off chance someone is using driverkit with `aarch64`.